internal/impl/kafka/integration_test.go

```diff
@@ -821,6 +822,188 @@ output:
 	}
 }
 
+func TestSchemaRegistryProtobufSchemasIntegration(t *testing.T) {
+	integration.CheckSkip(t)
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+	pool.MaxWait = time.Minute
+
+	sr, err := redpandatest.StartRedpanda(t, pool, false, true)
+	require.NoError(t, err)
+
+	t.Logf("Schema Registry URL: %s", sr.SchemaRegistryURL)
+
+	tests := []struct {
+		name       string
+		iterations int
+		normalize  bool
+		metadata   string
+		ruleSet    string
+	}{
+		{
+			name:       "allows creating the same schema twice",
+			iterations: 2,
+		},
+		{
+			name:       "normalises schemas",
+			iterations: 1,
+			normalize:  true,
+		},
+		{
+			name:       "removes metadata",
+			iterations: 1,
+			metadata: `{
+  "properties": {
+    "confluent:version": "1"
+  }
+}`,
+		},
+		{
+			name:       "removes rule sets",
+			iterations: 1,
+			ruleSet: `{
+  "domainRules": [
+    {
+      "name": "checkSsnLen",
+      "kind": "CONDITION",
+      "type": "CEL",
+      "mode": "WRITE",
+      "expr": "size(message.ssn) == 9"
+    }
+  ]
+}`,
+		},
+	}
+
+	writeSchema := func(t *testing.T, schema []byte, normalize, removeMetadata, removeRuleSet bool) {
+		streamBuilder := service.NewStreamBuilder()
+
+		// Set up a dummy `schema_registry` input which the output can connect to even though it won't need to fetch any
+		// schemas from it.
+		err = streamBuilder.AddInputYAML(fmt.Sprintf(`
+schema_registry:
+  url: %s
+  subject_filter: does_not_exist
+`, sr.SchemaRegistryURL))
+		require.NoError(t, err)
+
+		err = streamBuilder.AddOutputYAML(fmt.Sprintf(`
+schema_registry:
+  url: %s
+  subject: ${! json("subject") }
+  backfill_dependencies: true
+  normalize: %t
+  remove_metadata: %t
+  remove_rule_set: %t
+`, sr.SchemaRegistryURL, normalize, removeMetadata, removeRuleSet))
+		require.NoError(t, err)
+
+		prodFn, err := streamBuilder.AddProducerFunc()
+		require.NoError(t, err)
+
+		stream, err := streamBuilder.Build()
+		require.NoError(t, err)
+
+		doneChan := make(chan struct{})
+		go func() {
+			require.NoError(t, stream.Run(t.Context()))
+			close(doneChan)
+		}()
+		defer func() {
+			require.NoError(t, stream.StopWithin(3*time.Second))
+			<-doneChan
+		}()
+
+		require.NoError(t, prodFn(t.Context(), service.NewMessage(schema)))
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dummySubject := "foo"
+
+			defer func() {
+				// Clean up the subject at the end of each subtest.
+				deleteSubject(t, sr.SchemaRegistryURL, dummySubject, false)
+				deleteSubject(t, sr.SchemaRegistryURL, dummySubject, true)
+			}()
+
+			for range test.iterations {
+				dummyProtoSchema := `syntax = "proto3";
+package com.mycorp.mynamespace;
+
+message SampleRecord {
+  int32 my_field1 = 1;
+  double my_field2 = 2;
+  string my_field3 = 3;
+}`
+
+				dummyDenormalizedProtoSchema := `syntax = "proto3";
```

Comment:
How about a const. Also, please add a comment it took me a while to spot the difference.